output/formatter.go

```diff
@@ -91,12 +93,50 @@ func CreateReport(w io.Writer, format string, issues []*gosec.Issue, metrics *go
 		err = reportFromHTMLTemplate(w, html, data)
 	case "text":
 		err = reportFromPlaintextTemplate(w, text, data)
+	case "sonarqube":
+		err = reportSonarqube(rootPath, w, data)
 	default:
 		err = reportFromPlaintextTemplate(w, text, data)
 	}
 	return err
 }
 
+func reportSonarqube(rootPath string, w io.Writer, data *reportInfo) error {
+	var si sonarIssues
+	for _, issue := range data.Issues {
+		lines := strings.Split(issue.Line, "-")
+
+		startLine, _ := strconv.Atoi(lines[0])
+		endLine := startLine
+		if len(lines) > 1 {
+			endLine, _ = strconv.Atoi(lines[1])
+		}
+		s := sonarIssue{
+			EngineID: "gosec",
+			RuleID:   issue.RuleID,
+			PrimaryLocation: location{
+				Message:   issue.What,
+				FilePath:  strings.Replace(issue.File, rootPath+"/", "", 1),
+				TextRange: textRange{StartLine: startLine, EndLine: endLine},
+			},
+			Type:          "VULNERABILITY",
+			Severity:      getSonarSeverity(issue.Severity.String()),
+			EffortMinutes: 5,
+		}
+		si.SonarIssues = append(si.SonarIssues, s)
+	}
+	raw, err := json.MarshalIndent(si, "", "\t")
+	if err != nil {
+		panic(err)
```

Comment:
panic? could you return the error instead?

Reply:
That’s a copy and paste from the json formatter.

Reply:
I think we should probably return the error. I'll submit an issue to fix the json formatter and do so after this has been merged.

Reply:
error returned and fixed the json formatter.